clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/ChangeCaptureViewWriter.java

```diff
@@ -83,7 +85,15 @@ public Future<PubSubProduceResult> processRecord(
       initializeVeniceWriter(version);
     }
     // TODO: RecordChangeEvent isn't versioned today.
-    return veniceWriter.put(key, recordChangeEvent, 1);
+    return CompletableFuture.supplyAsync(() -> {
+      try {
+        return (PubSubProduceResult) veniceWriter.put(key, recordChangeEvent, 1).get();
```

Comment:
Hmm... why create a new CF which only does `.get()` of a `Future`? Is it to convert a `Future` into a `CF`? What is ultimately returned by this thing is `ApacheKafkaProducerCallback::getProduceResultFuture`, which is a function whose signature returns a `Future`, but whose implementation returns its `private final CompletableFuture<PubSubProduceResult> produceResultFuture` property. Maybe we should just change the signatures to return `CF` instead of `Future`, so this wrapping here isn't necessary?

Reply:
The signatures in veniceWriter?

Reply:
Yeah, this purely exists to just get the right object type :/

I just tweaked the veniceWriter interface.  Might make pulling it in internally a bit of a pain?  Not sure.

Reply:
We've gotta do what we've gotta do 🤷